### ISSUE TITLE
Configure various timeouts like scanning timeout, waiting for snapshots timeout, and iteration times

### ISSUE THREAD

**Issue by jeniawhite** [elastic/cloudbeat#740]
Make sure that all of the timeouts across the feature make sense and are compliant with what we want to achieve.
Example of this is the snapshot verifier:

```go
func (f VulnerabilityVerifier) verify(ctx context.Context, snap types.SnapshotInfo) {
	f.log.Info("Starting VulnerabilityVerifier.verify")
	// TODO: Change this to variable
	timer := time.After(5 * time.Minute)
	for {
		select {
		case <-timer:
			f.log.Info("VulnerabilityVerifier.verify timed out waiting for snapshot ", *snap.SnapshotId)
			return
		case <-ctx.Done():
			f.log.Info("VulnerabilityVerifier.verify context canceled")
			return
		// TODO: Change this to variable
		case <-time.After(15 * time.Second):
			sp, err := f.provider.DescribeSnapshots(ctx, snap)
			if err != nil {
				f.log.Errorf("VulnerabilityVerifier.verify.DescribeSnapshots failed: %v", err)
				continue
			}
			// TOOD: Add a layer of "smart" cache to avoid checking and sending the same snapshot
			// and not to wait on all snapshots to be completed, sending them periodically
			allCompleted := true
			for _, i := range sp {
				if i.State != types.SnapshotStateCompleted {
					f.log.Info("VulnerabilityVerifier.verify.VerifySnapshot snapshot not completed yet - ", *i.SnapshotId)
					allCompleted = false
					break
				}
			}
			if allCompleted {
				for _, i := range sp {
					f.ch <- i
				}
				return
			}
		}
	}
}
```

You can see comments on timeouts like `// TODO: Change this to variable`.
Another place is the scanning of the EBS snapshots itself by `Trivy` this is also capped:

```go
		GlobalOptions: flag.GlobalOptions{
			// TODO: Make configurable
			Timeout: 1 * time.Hour,
			Quiet:   false,
			Debug:   true,
			// CacheDir: "/tmp/trivy",
		},
```
In this case comment on timeout `// TODO: Make configurable`.
We would like to evaluate the times there and make sure that they are making sense.

**Comment by uri-weisman** [comment]
I have generated a report that includes data on snapshot creation time and scanning.
I ignored the data on encrypted snapshots that we do not currently support, it appears that we are experiencing an issue with the current snapshot creation timeout, which is set at **10 mins**. 
Several non-encrypted snapshots took longer than this threshold to be created, with the longest one taking 26 mins, there doesn't seem to be a correlation with snapshot sizes, as volume sizes varied from 8 to 80 GBs.
Since AWS controls snapshot creation, we have limited control over it, therefore, I recommend extending the timeout to **30 mins**.
Regarding the scanning timeout, which is currently set at 1 hour, there is no problem as no snapshot reached this limit.


Report - [timeout.xlsx](https://github.com/elastic/cloudbeat/files/11380034/timeout.xlsx)
Code to generate this report - https://github.com/uri-weisman/cloudbeat/tree/verify-timeouts

**Comment by uri-weisman** [comment]
@oren-zohar- Can we close this and create a new ticket regarding encrypted snapshots?

**Comment by oren-zohar** [comment]
> @oren-zohar- Can we close this and create a new ticket regarding encrypted snapshots?

sounds good to me :+1:

**Comment by uri-weisman** [comment]
Opened a follow up task for encrypted snapshots
https://github.com/elastic/cloudbeat/issues/936
